swarm/src/lib.rs

```diff
@@ -624,6 +633,31 @@ where TBehaviour: NetworkBehaviour,
         self
     }
 
+    /// Sets the executor to use to spawn background tasks.
+    ///
+    /// By default, uses a threads pool.
+    pub fn executor(mut self, executor: impl Spawn + 'static) -> Self {
+        self.executor = Some(Box::new(executor));
+        self
+    }
+
+    /// Shortcut for calling `executor` with an object that calls the given closure.
+    pub fn executor_fn(mut self, executor: impl Fn(FutureObj<'static, ()>) + 'static) -> Self {
+        use futures::task::SpawnError;
+        struct SpawnImpl<F>(F);
+        impl<F: Fn(FutureObj<'static, ()>)> Spawn for SpawnImpl<F> {
+            fn spawn_obj(&self, future: FutureObj<'static, ()>) -> Result<(), SpawnError> {
+                (self.0)(future);
+                Ok(())
+            }
+            fn status(&self) -> Result<(), SpawnError> {
+                Ok(())
+            }
```

Comment:
This is the default impl, so not really necessary, unless you prefer being explicit here.